src/pip/_internal/utils/logging.py

```diff
@@ -44,15 +44,23 @@ def get_indentation():
 
 
 class IndentingFormatter(logging.Formatter):
+    def __init__(self, *args, **kwargs):
```

Comment:
I think it would be worth adding a docstring here and documenting the kwarg you're adding (and default behavior), since it doesn't appear in the function signature and because it's used in the config.

Also, I would consider renaming the argument (and attribute name) to something like "with_timestamp" or "add_timestamp" since timestamp is a noun and makes it sound like the timestamp itself is being passed.